test/base/test_buffer.py

```diff
@@ -278,6 +280,48 @@ def test_pickle():
                        pbuf.weight[np.arange(len(pbuf))])
 
 
+def test_hdf5():
+    size = 100
+    vbuf = ReplayBuffer(size, stack_num=2)
+    device = 'cuda' if torch.cuda.is_available() else 'cpu'
+    rew = torch.tensor([1.]).to(device)
+    for i in range(4):
+        vbuf.add(
+                obs=Batch(index=np.array([i])),
+                act=i,
+                rew=rew,
+                done=0,
+                info={"number": i})
+
+    # save
+    f, path = tempfile.mkstemp(suffix='.hdf5')
+    os.close(f)
+    vbuf.save(path)
+
+    # load replay buffer
+    _vbuf = ReplayBuffer.load(path)
+
+    def assertions(vbuf, _vbuf):
+        assert len(_vbuf) == len(vbuf) and np.allclose(_vbuf.act, vbuf.act)
+        assert _vbuf.stack_num == vbuf.stack_num
+        assert _vbuf._maxsize == vbuf._maxsize
+        assert _vbuf._index == vbuf._index
+        assert np.all(_vbuf._indices == vbuf._indices)
+        assert isinstance(vbuf.get(0, "info"), Batch)
+        assert isinstance(_vbuf.get(0, "info"), Batch)
+        assert np.all(vbuf[:]["info"]["number"] == _vbuf[:]["info"]["number"])
+
+    assertions(vbuf, _vbuf)
+
+    # load only contents of replay buffer
+    _vbuf = ReplayBuffer(size, stack_num=2)
+    _vbuf.load_contents(path)
+
+    assertions(vbuf, _vbuf)
+
+    os.remove(path)
+
+
 if __name__ == '__main__':
     test_replaybuffer()
     test_ignore_obs_next()
```

Comment:
You can also add test_hdf5 here. (But it doesn't matter in the ci)

Reply:
I have added the version requirement h5py>=3.1.0 because this is the version I have tested the code with. Possibly runs with older versions as well though. I also added the test_hdf5 and removed the swap files as you suggested.